src/darksend.cpp

```diff
@@ -1763,7 +1763,32 @@ bool CDarksendPool::CreateDenominated(int64_t nTotalValue)
     }
 
     // ****** Add denoms ************ /
+
+    std::vector<CTxIn> vTempDenomCoins;
+    std::vector<COutput> vTempDenomCoins2;
+    int64_t nValueIn = 0;
+
     BOOST_REVERSE_FOREACH(int64_t v, darkSendDenominations){
+
+        // Note: denoms are skipped if there are already DENOMS_COUNT_MAX of them
+
+        // check skipped denoms
+        if(IsDenomSkipped(v)) continue;
+
+        int denombitshifted = GetDenominationsByAmount(v);
+
+        nValueIn = 0;
+        vTempDenomCoins.clear();
+        vTempDenomCoins2.clear();
+
+        // find new denoms to skip if any (ignore the largest one)
+        if (v != darkSendDenominations[0] && pwalletMain->SelectCoinsByDenominations(denombitshifted, v*DENOMS_COUNT_MAX, MAX_MONEY, vTempDenomCoins, vTempDenomCoins2, nValueIn, 0, nDarksendRounds+1)){
```

Comment:
How about (pwalletMain->CountInputsWithAmount(v) > DENOMS_COUNT_MAX) , is this enough?, it's look much easier.